### ISSUE TITLE
Add tests to follow PEP8 guidelines

### ISSUE THREAD

**Issue by devkapilbansal** [anitab-org/bridge-in-tech-backend#226]
### Is your feature request related to a problem? Please describe.
PEP stands for Python Enhancement Proposal. It is a style document that documents some best practices

### Describe the solution you'd like
Code written in Python should follow standard practices and PEP8 guidelines. For this a linter that follows these guidelines or some sort of checks can be implemented

### Describe alternatives you've considered
I would suggest using Flake8 to test if PEP8 best practices are followed in the code written or not. Additionally, Flake8 workflow tests can be setup to ensure pull requests made follow these guidelines as well.

### Additional context
Flake8 is not the only tool available and other alternatives can be considered.

**Comment by Aryamanz29** [comment]
I would like to work on this one @devkapilbansal

**Comment by decon-harsh** [comment]
> I would like to work on this one @devkapilbansal 

You have been already assigned one issue. Please claim it after you send a successful PR for the former.

**Comment by Aryamanz29** [comment]
@decon-harsh PR raised  #251

**Comment by devkapilbansal** [comment]
@Aryamanz29 are you working on this?

**Comment by Aryamanz29** [comment]
> @Aryamanz29 are you working on this?

Yes,
Sorry for the delay I'll raise PR soon.

**Comment by Aryamanz29** [comment]
@devkapilbansal  While testing my code locally most `unittest` fails and I'm unable to debug it after several attempts, So kindly unassign this issue to me for now. I'll try to work on this issue later.


![Screenshot (49)](https://user-images.githubusercontent.com/56113566/114217750-30e65180-9986-11eb-9898-8388c708c193.png)

**Comment by epicadk** [comment]
Are you using @mtreacy002's forked repo as the Mentorship backend?

**Comment by mtreacy002** [comment]
@Aryamanz29 , can you check if running the test from the Test Explorer of your VS Code IDE pass? I've asked since this is also happened to me when I'm using Windows OS (running tests inside the terminal failed but running them inside VS Code IDE Test explorer passed). I have opened an issue #133 to investigate this. ATM when using Windows OS, I just run the tests on the VS Code terminal. On MacOS this issue doesn't exist (both terminal and VS Code IDE work just fine). cc @epicadk

**Comment by Aryamanz29** [comment]
@mtreacy002  Test fails same as they on the terminal, Now I'll try to set up this repo on my Linux machine to solve this problem😅

![Screenshot (51)](https://user-images.githubusercontent.com/56113566/114258901-9fa7c700-99e7-11eb-95d7-ec6615e7cf11.png)

**Comment by mtreacy002** [comment]
@Aryamanz29 , also, have you completed the compulsory #244 before working on this issue? You shouldn’t face this issue once you’ve successfully completed the task there. cc @epicadk

**Comment by epicadk** [comment]
> @Aryamanz29 , also, have you completed the compulsory #244 before working on this issue? You shouldn’t face this issue once you’ve successfully completed the task there. cc @epicadk

I guess this was before I was your message 😅. @Aryamanz29 please complete #244  before you attempt this.

**Comment by Aryamanz29** [comment]
> > @Aryamanz29 , also, have you completed the compulsory #244 before working on this issue? You shouldn’t face this issue once you’ve successfully completed the task there. cc @epicadk
> 
> I guess this was before I was your message 😅. @Aryamanz29 please complete #244 before you attempt this.

I have raised PR #267  for env setup, Please have a look to it @epicadk

**Comment by mtreacy002** [comment]
@devkapilbansal and @epicadk , since we're running lint workflow using black, wouldn't this pr (using Flake8) contradicting the workflow. I mean the 2 (black and flake8) don't always play nice with each other 🤔 . Unless you're thinking of using [flake8-black](https://pypi.org/project/flake8-black/) plugin, perhaps on the VSCode IDE? But I still don't see why we need to add flake8 workflow on top of black linting 🤔

**Comment by devkapilbansal** [comment]
Hi @mtreacy002 black is used to lint the code so to increase readability but flake8 ensures that the code follows Python standards specified by PEP8. 
There are some cases where black and flake8 contradicts each other but I think that can be managed using config file.

**Comment by mtreacy002** [comment]
> Hi @mtreacy002 black is used to lint the code so to increase readability but flake8 ensures that the code follows Python standards specified by PEP8.
> There are some cases where black and flake8 contradicts each other but I think that can be managed using config file.

May I ask what you think of the flake8-black plugin I mentioned above? wouldn't this be a faster and easier option instead of writing the config file, @devkapilbansal ?

**Comment by devkapilbansal** [comment]
Yes, it looks good to. I was unaware of this. I think we can proceed with this instead of having flake8 and flask.
